crates/ruff_db/src/file_system.rs

```diff
@@ -0,0 +1,265 @@
+use std::fmt::Formatter;
+use std::ops::Deref;
+use std::path::Path;
+
+use camino::{Utf8Path, Utf8PathBuf};
+use filetime::FileTime;
+
+pub use memory::MemoryFileSystem;
+pub use os::OsFileSystem;
+
+mod memory;
+mod os;
+
+pub type Result<T> = std::io::Result<T>;
```

Comment:
It makes it easier to change the result type later, because I made sure that all methods return `file_system::Result`. It's also easier to write because I don't have to think about what's the right return type. I can just use that.